frontend/packages/modelServing/modelRegistry/useDeployButtonState.ts

```diff
@@ -0,0 +1,40 @@
+import { SupportedArea } from '@odh-dashboard/internal/concepts/areas/types';
+import useIsAreaAvailable from '@odh-dashboard/internal/concepts/areas/useIsAreaAvailable';
+import { DEPLOY_BUTTON_TOOLTIP } from '@odh-dashboard/internal/pages/modelServing/screens/const';
+import { useAvailableClusterPlatforms } from '../src/concepts/useAvailableClusterPlatforms';
+
+const KSERVE_ID = 'kserve';
+
+/**
+ * Returns the deploy button state (visible, enabled, tooltip) for model deploy actions.
+ */
+const useDeployButtonState = (): { visible: boolean; enabled?: boolean; tooltip?: string } => {
+  const isModelServingEnabled = useIsAreaAvailable(SupportedArea.MODEL_SERVING).status;
+  const { clusterPlatforms } = useAvailableClusterPlatforms();
+  const kServe = clusterPlatforms.find((p) => p.properties.id === KSERVE_ID);
```

Comment:
I agree with Griffin here that you're introducing i knowledge of kserve in the base component that should not know about kserve.